pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -0,0 +1,333 @@
+			// In case of not found, it means the object is gone but informer cache is not updated yet, we can safely delete it from our store
+			if err != nil && errors.IsNotFound(err) {
+				c.store.UnlockDelete(key, c.ID)
+				return autoscaling.NoRequeue, nil
+			}
+
+			// In all other cases, we requeue and wait for the object to be deleted from store with next reconcile
+			c.store.Unlock(key)
+			return autoscaling.Requeue, err
+		}
+
+		// If the object is owned by remote config and newer, we need to update the spec in Kubernetes
+		// If Kubernetes is newer, we wait for RC to update the object in our internal store.
+		if *podAutoscalerInternal.SettingsVersion > podAutoscaler.Spec.RemoteVersion {
+			podAutoscaler.Spec = podAutoscalerInternal.Spec
+			err := c.updatePodAutoscalerSpec(ctx, podAutoscalerInternal, podAutoscaler)
+
+			// If update succeeded, we requeue the object to handle status update
+			if err == nil {
+				c.Workqueue.AddRateLimited(key)
+			}
+
+			// When doing an external update, we stop and reqeue the object to not have multiple changes at once.
+			c.store.Unlock(key)
+			return autoscaling.Requeue, err
+		}
+
+		// TODO: If Generation != podAutoscaler.Generation, we should compute `.Spec` hash
+		// and compare it with the one in the PodAutoscaler. If they differ, we should update the PodAutoscaler.
+	}
+
+	// Implement sync logic for local ownership, source of truth is Kubernetes
+	if podAutoscalerInternal.Spec.Owner == datadoghq.DatadogPodAutoscalerLocalOwner {
+		if podAutoscalerInternal.Generation != podAutoscaler.Generation {
+			podAutoscalerInternal.UpdateFromSpec(&podAutoscaler.Spec)
+			c.store.Set(key, podAutoscalerInternal, c.ID)
+		}
+	}
+
+	// Now that everything is synced, we can perform the actual processing
+	result, err := c.handleScaling(ctx, &podAutoscalerInternal)
+	if err != nil {
+		podAutoscalerInternal.GlobalError = err
+		result.updateStatus = true
+	}
+
+	if result.updateStatus {
+		statusErr := c.updatePodAutoscalerStatus(ctx, podAutoscalerInternal, podAutoscaler)
+		if statusErr != nil {
+			log.Errorf("Failed to update status for PodAutoscaler: %s/%s, err: %v", ns, name, statusErr)
+		}
+	}
+
+	c.store.UnlockSet(key, podAutoscalerInternal, c.ID)
+	return result.ProcessResult, err
+}
+
+func (c *Controller) handleScaling(ctx context.Context, podAutoscalerInternal *model.PodAutoscalerInternal) (processResult, error) {
+	// First, resolve target GVK if not already done
+	if podAutoscalerInternal.TargetGVK.Empty() {
+		gv, err := schema.ParseGroupVersion(podAutoscalerInternal.Spec.TargetRef.APIVersion)
+		if err != nil {
+			return withStatusUpdate(true, autoscaling.NoRequeue), fmt.Errorf("failed to parse API version %s: %w", podAutoscalerInternal.Spec.TargetRef.APIVersion, err)
+		}
+
+		podAutoscalerInternal.TargetGVK = schema.GroupVersionKind{
+			Group:   gv.Group,
+			Version: gv.Version,
+			Kind:    podAutoscalerInternal.Spec.TargetRef.Kind,
+		}
+	}
```

Comment:
I think we can make the parsing of TargetGVK a method of `podAutoscalerInternal`.